File: hcesdk/src/main/java/com/conduent/hcesdk/entities/remoteoffer/response/RemoteResponse.java
package com.conduent.hcesdk.entities.remoteoffer.response;

import com.conduent.hcesdk.entities.valuesapi.ProductCustomData;

import java.util.ArrayList;

public class RemoteResponse {
    private String ItemCategory;

    private String TitleOffer;

    private int UnitPrice;

    private Counter Counter;

    private String RequiresAuth;

    private ProductCustomData CustomData;

    private ArrayList<DatesZonesPrices> DatesZonesPrices;

    private int ArticleID;

    private int ProductID;

    private int NetworkID;

    private int VatRate;

    public String getItemCategory ()
    {
        return ItemCategory;
    }

    public void setItemCategory (String ItemCategory)
    {
        this.ItemCategory = ItemCategory;
    }

    public String getTitleOffer ()
    {
        return TitleOffer;
    }

    public void setTitleOffer (String TitleOffer)
    {
        this.TitleOffer = TitleOffer;
    }

    public int getUnitPrice ()
    {
        return UnitPrice;
    }

    public void setUnitPrice (int UnitPrice)
    {
        this.UnitPrice = UnitPrice;
    }

    public Counter getCounter ()
    {
        return Counter;
    }

    public void setCounter (Counter Counter)
    {
        this.Counter = Counter;
    }

    public String getRequiresAuth ()
    {
        return RequiresAuth;
    }

    public void setRequiresAuth (String RequiresAuth)
    {
        this.RequiresAuth = RequiresAuth;
    }

    public ProductCustomData getCustomData ()
    {
        return CustomData;
    }

    public void setCustomData (ProductCustomData CustomData)
    {
        this.CustomData = CustomData;
    }

    public ArrayList<DatesZonesPrices> getDatesZonesPrices ()
    {
        return DatesZonesPrices;
    }

    public void setDatesZonesPrices (ArrayList<DatesZonesPrices> DatesZonesPrices)
    {
        this.DatesZonesPrices = DatesZonesPrices;
    }

    public int getArticleID ()
    {
        return ArticleID;
    }

    public void setArticleID (int ArticleID)
    {
        this.ArticleID = ArticleID;
    }

    public int getProductID ()
    {
        return ProductID;
    }

    public void setProductID (int ProductID)
    {
        this.ProductID = ProductID;
    }

    public int getNetworkID ()
    {
        return NetworkID;
    }

    public void setNetworkID (int NetworkID)
    {
        this.NetworkID = NetworkID;
    }

    public int getVatRate ()
    {
        return VatRate;
    }

    public void setVatRate (int VatRate)
    {
        this.VatRate = VatRate;
    }

    @Override
    public String toString()
    {
        return "ClassPojo [ItemCategory = "+ItemCategory+", TitleOffer = "+TitleOffer+", UnitPrice = "+UnitPrice+", Counter = "+Counter+", RequiresAuth = "+RequiresAuth+", CustomData = "+CustomData+", DatesZonesPrices = "+DatesZonesPrices+", ArticleID = "+ArticleID+", ProductID = "+ProductID+", NetworkID = "+NetworkID+", VatRate = "+VatRate+"]";
    }
}

File: hcesdk/src/main/java/com/conduent/hcesdk/entities/remoteoffer/response/RemoteResponse.kt
package com.conduent.hcesdk.entities.remoteoffer.response

import com.google.gson.annotations.SerializedName

data class RemoteResponse(
    @SerializedName("ArticleID") val articleID : Int,
    @SerializedName("Counter") val counter : Counter,
    @SerializedName("DatesZonesPrices") val datesZonesPrices : List<DatesZonesPrices>,
    @SerializedName("ItemCategory") val itemCategory : String,
    @SerializedName("NetworkID") val networkID : Int,
    @SerializedName("ProductID") val productID : Int,
    @SerializedName("TitleOffer") val titleOffer : String,
    @SerializedName("UnitPrice") val unitPrice : Int,
    @SerializedName("VatRate") val vatRate : Int,
    @SerializedName("CustomData") val customData : CustomData,
    @SerializedName("RequiresAuth") val requiresAuth : String

)

data class Counter (

    @SerializedName("MaxBuyValue") val maxBuyValue : Int,
    @SerializedName("MinBuyValue") val minBuyValue : Int,
    @SerializedName("StepBuyValue") val stepBuyValue : Int
)

data class CustomData (

    @SerializedName("CustomData1") val customData1 : String,
    @SerializedName("CustomData2") val customData2 : String,
    @SerializedName("CustomData3") val customData3 : String,
    @SerializedName("CustomData4") val customData4 : String
)

data class DatesZonesPrices (

    @SerializedName("Dates") val dates : List<DDates>,
    @SerializedName("ZonesPrices") val zonesPrices : List<ZonesPrices>
)

data class DDates (

    @SerializedName("Duration") val duration : Int,
    @SerializedName("MaxStartDate") val maxStartDate : String,
    @SerializedName("MinStartDate") val minStartDate : String,
    @SerializedName("Step") val step : Int,
    @SerializedName("Unit") val unit : String
)

data class ZonesPrices (

    @SerializedName("UnitPrice") val unitPrice : Int,
    @SerializedName("VatRate") val vatRate : Int,
    @SerializedName("ZoneID") val zoneID : String,
    @SerializedName("ZoneLabel") val zoneLabel : String
)